question16/main.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const createElementWithClassName = (element, name) => {
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
+const newsWrapper = document.getElementById('js-newsWrapper');
+const tabsGroup = document.getElementById('js-tabs');
+const contentsWrapper = document.createElement("div");
+contentsWrapper.id = "js-contentsWrapper";
+contentsWrapper.classList.add("contentsWrapper");
+
+
+//Get json data
+async function callApi() {
+  try {
+    const res = await fetch('https://myjson.dit.upm.es/api/bins/an1r');
+    const json = await res.json();
+    return json.data;
+  }
+  catch(error) {
+    tabsGroup.textContent = "データが取得できませんでした";
+    console.error(error);
+  }
+  finally {
+    console.log('CONPLETE!');
+  }
+}
+
+async function init() {
+  const result = await callApi();
+  if (result){
+    renderElement(result);
+    } else {
+    tabsGroup.textContent = "データの表示に失敗しました";
+    console.error('データの表示に失敗しました'); 
+  }
+}
+init();
+
+//Rendering tabs and contents
+function renderElement(data){
+    newsWrapper.appendChild(contentsWrapper);
+    const length = data.length;
+    const fragmentTablists = document.createDocumentFragment();
+    const fragmentContents = document.createDocumentFragment();
+
+    //Display category names in tabs
+    for (let i = 0; i < length; i++){
+      const {category, articles, img} = data[i];
+      const tabLists = document.createElement('li');
+     
+      tabLists.innerHTML = category;
```

Comment:
Since the type of `category` is string, it may be better to use textContent instead of innerHTML.

Reply:
You're right! I fixed to use use textContent instead of innerHTML.  https://github.com/csakyo/homework/pull/29/commits/583f0fed25f0a743b7d25f48f6131dfe731f897a

And I just realized there was using innerHTML other place so I fixed it. 
https://github.com/csakyo/homework/pull/29/commits/781dbffd143d757175968f8f9890c85bb1884295